kpi/tests/api/v2/test_api_assets.py

```diff
@@ -311,6 +317,42 @@ def test_asset_version_id_and_content_hash(self):
         self.assertEqual(response.data['version__content_hash'],
                          self.asset.latest_version.content_hash)
 
+    def test_submission_count(self):
+        anotheruser = User.objects.get(username='anotheruser')
+        self.asset.deploy(backend='mock', active=True)
+        submissions = [
+            {
+                "__version__": self.asset.latest_deployed_version.uid,
+                "q1": "a1",
+                "q2": "a2",
+                "id": 1,
+                "_submitted_by": ""
+            },
+            {
+                "__version__": self.asset.latest_deployed_version.uid,
+                "q1": "a3",
+                "q2": "a4",
+                "id": 2,
+                "_submitted_by": anotheruser.username
+            }
+        ]
+
+        self.asset.deployment.mock_submissions(submissions)
+        partial_perms = {
+            PERM_VIEW_SUBMISSIONS: [{
+                '_submitted_by': anotheruser.username
+            }]
+        }
+        self.asset.assign_perm(anotheruser, PERM_PARTIAL_SUBMISSIONS,
+                               partial_perms=partial_perms)
+
+        response = self.client.get(self.asset_url, format='json')
+        self.assertEqual(response.data['deployment__submission_count'], 2)
+        self.client.logout()
+        self.client.login(username='anotheruser', password='anotheruser')
+        response = self.client.get(self.asset_url, format='json')
+        self.assertEqual(response.data['deployment__submission_count'], 1)
```

Comment:
Ok,
The missing leading underscore is because of the `MockDeploymentBackend` which looks for `id`, not `_id`.
I will change it to be consistent with `Mongo` and `KobocatDeploymentBackend`.